### PR TITLE
SAVB-31 Regression testing should be added

### PR DESCRIPTION
**Pull request by tvsonar**

### CODE REVIEW

File: src/SonarLint.CSharp/Helpers/SymbolicExecution/ExplodedGraph.cs [hunk 1]
```diff
@@ -374,7 +374,9 @@ protected override void VisitInstruction(ExplodedGraphNode node)
                     break;
 
                 case SyntaxKind.ImplicitElementAccess:
-                    newProgramState = newProgramState.PopValues(((ImplicitElementAccessSyntax)instruction).ArgumentList?.Arguments.Count ?? 0);
+                    newProgramState = newProgramState
+                        .PopValues(((ImplicitElementAccessSyntax)instruction).ArgumentList?.Arguments.Count ?? 0)
+                        .PushValue(new SymbolicValue());
```
Comment (m-g-sonar) [inline src/SonarLint.CSharp/Helpers/SymbolicExecution/ExplodedGraph.cs]: To keep consistency with other cases from the switch, it would be better to separate these 2 chained method calls into 2 distinct statements (1 to pop, 1 to push).

Comment (tvsonar) [inline src/SonarLint.CSharp/Helpers/SymbolicExecution/ExplodedGraph.cs]: I think new code should use this format, I was trying to use it lately, but not changing all usages at once.

Comment (m-g-sonar) [inline src/SonarLint.CSharp/Helpers/SymbolicExecution/ExplodedGraph.cs]: Okay, then don't change anything now, but it would be nice to changing everything in another commit soon.
